File: Projects/TP06AdapterView/app/src/main/java/com/hyun/tp06adapterview/MyAdapter.java
package com.hyun.tp06adapterview;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH> {

    ArrayList<Item> items;
    Context context;

    public MyAdapter(ArrayList<Item> items, Context context) {
        this.items = items;
        this.context = context;
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater= LayoutInflater.from(context);
        View itemView= inflater.inflate(R.layout.recycler_item,parent,false);

        VH holder= new VH(itemView);
        return holder;


    }


    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {

        Item item= items.get(position);

        holder.iv.setImageResource(item.img);
        holder.name.setText(item.name);
        holder.iv.setImageResource(item.img);


    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class VH extends RecyclerView.ViewHolder{
        ImageView iv;
        TextView name;
        TextView lv;

        public VH(@NonNull View itemView) {
            super(itemView);

            iv= itemView.findViewById(R.id.iv);
            name= itemView.findViewById(R.id.name);
            lv= itemView.findViewById(R.id.lv);

        }
    }



    }

File: Projects/Ex60ViewBinding/app/src/main/java/com/hyun/ex60viewbinding/MyAdapter.kt
package com.hyun.ex60viewbinding

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.hyun.ex60viewbinding.databinding.RecyclerItemBinding

class MyAdapter constructor(val context:Context, val itemList:MutableList<Item>) : Adapter<MyAdapter.VH>(){

    //아이템 뷰 1개의 자식뷰들의 참조변수를 저장하고 있는 ViewHolder 클래스
    inner class VH constructor(itemView:View) : ViewHolder(itemView){
        //자식뷰 참조변수를 모두 만들지 말고.. 바인딩 객체 참조변수 1개만 보유
        lateinit var binding: RecyclerItemBinding
        init {
            //이미 뷰객체가 만들어져 있다면..그냥 바인딩 객체와 연결만 하면...자식뷰 참조값을 모두 보유하게 됨
            binding= RecyclerItemBinding.bind(itemView)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val itemView:View = LayoutInflater.from(context).inflate(R.layout.recycler_item, parent, false)
        return VH(itemView)
    }

    override fun getItemCount(): Int {
        return itemList.size
    }

    override fun onBindViewHolder(holder: VH, position: Int) {
        holder.binding.iv.setImageResource(itemList[position].imgId)
        holder.binding.tv.text= itemList[position].title
    }

}